From 74dd531e3016d139532d830d1c14c383ba570662 Mon Sep 17 00:00:00 2001
From: Takahiko Saito <tysaito@gmail.com>
Date: Wed, 16 Sep 2015 15:34:00 -0800
Subject: [PATCH 0298/1164] CDH-31826 : HIVE-11820 : export tables with size
 of >32MB throws
 java.lang.IllegalArgumentException: Skip CRC is
 valid only with update options (Takahiko Saito
 via Ashutosh Chauhan)

Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>

Conflicts:
	shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
---
 .../apache/hadoop/hive/shims/Hadoop20SShims.java   |    5 ++++-
 .../apache/hadoop/hive/shims/Hadoop23Shims.java    |    3 ++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/shims/0.20S/src/main/java/org/apache/hadoop/hive/shims/Hadoop20SShims.java b/shims/0.20S/src/main/java/org/apache/hadoop/hive/shims/Hadoop20SShims.java
index 8575567..fe2b1cb 100644
--- a/shims/0.20S/src/main/java/org/apache/hadoop/hive/shims/Hadoop20SShims.java
+++ b/shims/0.20S/src/main/java/org/apache/hadoop/hive/shims/Hadoop20SShims.java
@@ -70,6 +70,8 @@
 import org.apache.hadoop.security.UserGroupInformation;
 import org.apache.hadoop.tools.distcp2.DistCp;
 import org.apache.hadoop.tools.distcp2.DistCpOptions;
+import org.apache.hadoop.tools.distcp2.DistCpOptions.FileAttribute;
+
 import org.apache.hadoop.util.Progressable;
 import org.apache.hadoop.util.Tool;
 import org.apache.hadoop.util.VersionInfo;
@@ -664,8 +666,9 @@ public String getShortName() throws IOException {
   public boolean runDistCp(Path src, Path dst, Configuration conf) throws IOException {
 
     DistCpOptions options = new DistCpOptions(Collections.singletonList(src), dst);
-    options.setSkipCRC(true);
     options.setSyncFolder(true);
+    options.setSkipCRC(true);
+    options.preserve(FileAttribute.BLOCKSIZE);
     try {
       DistCp distcp = new DistCp(conf, options);
       distcp.execute();
diff --git a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
index 7a84a63..6d8f285 100644
--- a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
+++ b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
@@ -88,6 +88,7 @@
 import org.apache.hadoop.security.UserGroupInformation;
 import org.apache.hadoop.tools.DistCp;
 import org.apache.hadoop.tools.DistCpOptions;
+import org.apache.hadoop.tools.DistCpOptions.FileAttribute;
 import org.apache.hadoop.util.Progressable;
 import org.apache.hadoop.util.Tool;
 import org.apache.hadoop.yarn.conf.YarnConfiguration;
@@ -1141,7 +1142,7 @@ public boolean runDistCp(Path src, Path dst, Configuration conf) throws IOExcept
     DistCpOptions options = new DistCpOptions(Collections.singletonList(src), dst);
     options.setSyncFolder(true);
     options.setSkipCRC(true);
-    options.preserve(DistCpOptions.FileAttribute.BLOCKSIZE);
+    options.preserve(FileAttribute.BLOCKSIZE);
     try {
       DistCp distcp = new DistCp(conf, options);
       distcp.execute();
-- 
1.7.9.5

